src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
```

Comment:
Honestly I don't love the name. Also, I'm not sure if what we should do instead is to have this code be part of the UnixDriver such that we pick this driver's implementation whenever kernel is greater than 4.8 and fall back to sysfs otherwise.

Reply:
Agree, but we have discussed to leave runtime selection for now, if you want to include that with this PR we need to discuss how exactly do that

Reply:
I'm not sure if making it internal is what we should do also, consider this case: there is a bug in our libgpiodDriver and people want to force the use of the sysfsDriver even when their kernel is higher than 4.8. We could also expose an enum on a factory method to solve this, but my thinking for the unix genereic driver is the following:

- We should have three classes: UnixDriver(this will be an abstract class), SysFSDriver(implements UnixDriver class), and LibGpiodDriver(also implements UnixDriver class).
- UnixDriver abstract class will have a factory method which will by default pick either the sysfs driver or the LibGpiodDriver. This factory method will be called when initializing a controller and we decided to pick the generic driver.
- UnixDriver will also provide a way (perhaps an enum or something) to force picking the sysfs or the Libgpiod driver if desired.